doc/.gitignore

```diff
@@ -0,0 +1,4 @@
+# generated
```

Comment:
What's the intuition of adding an extra `.gitignore` to the `doc` directory?

Reply:
Generating the documentation leaves files that should not be tracked be git. Pretty standard.